realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -45,6 +45,10 @@ jclass java_lang_string;
 jmethodID java_lang_double_init;
 jclass java_util_date;
 jmethodID java_util_date_init;
+#if REALM_ENABLE_SYNC
+jclass java_syncmanager_class;
```

Comment:
It's the same situation in https://github.com/realm/realm-java/blob/abf24d36e3354253ba2742c9a36252b9edf1b56e/realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp#L83-L106

you do the lookup (and cache using `static`) in the VM thread, then capture it by copy in the lambda executed in C++